src/GovUk.Education.ExploreEducationStatistics.Content.Services/DataSetService.cs

```diff
@@ -34,34 +36,42 @@ public DataSetService(ContentDbContext contentDbContext)
         Guid? themeId = null,
         Guid? publicationId = null,
         Guid? releaseId = null,
-        bool? latest = true,
+        bool? latestOnly = true,
         string? searchTerm = null,
         DataSetsListRequestOrderBy? orderBy = null,
         SortOrder? sort = null,
         int page = 1,
-        int pageSize = 10)
+        int pageSize = 10,
+        CancellationToken cancellationToken = default)
     {
+        // If latestOnly is not specified default it to true except when a releaseId is specified
+        latestOnly ??= !releaseId.HasValue;
+
         orderBy ??= searchTerm == null ? Title : Relevance;
         sort ??= orderBy is Title or Natural ? Asc : Desc;
 
+        var latestPublishedReleaseVersions =
+            _contentDbContext.Releases.LatestReleaseVersions(publicationId, publishedOnly: true);
+
```

Comment:
I'd be inclined to move this inside `HavingLatestPublishedReleaseVersions` if possible. 

I also think right now we may be making this query unnecessarily, if `latestOnly` is true - although I don't consider that a big deal.

Reply:
The value `latestPublishedReleaseVersions` is not a query, it's a queryable. It hasn't been executed yet so the worst that's happened unnecessarily is the query tree expression has been built.

I considered putting it within  `HavingLatestPublishedReleaseVersions` but that involves passing the `ContentDbContext` so I decided against that.